models/DataObject/ClassDefinition/Data/Time.php

```diff
@@ -113,16 +113,16 @@ public function checkValidity($data, $omitMandatoryCheck = false, $params = [])
             throw new Model\Element\ValidationException('Wrong time format given must be a 5 digit string (eg: 06:49) [ '.$this->getName().' ]');
         }
 
-        if (!$omitMandatoryCheck && strlen($data)) {
+        if (!$omitMandatoryCheck && is_string($data) && strlen($data)) {
             if (!$this->toTime($data)) {
                 throw new Model\Element\ValidationException('Wrong time format given must be a 5 digit string (eg: 06:49) [ '.$this->getName().' ]');
             }
 
-            if (strlen($this->getMinValue()) && $this->isEarlier($this->getMinValue(), $data)) {
+            if (is_string($this->getMinValue()) && strlen($this->getMinValue()) && $this->isEarlier($this->getMinValue(), $data)) {
```

Comment:
```suggestion
            if ($this->getMinValue() && $this->isEarlier($this->getMinValue(), $data)) {
```
Should be `string|null`, so I think it is enough to check it without `strlen()`

Reply:
Wouldn't that cause a different behavior for falsy values like "0"?

Reply:
@sdrewergutland minValue/maxValue should be the same string schema like the data. The setters always set a string with 5 digit (eg: 06:49). So I think "0" is not allowed.

Reply:
Yep, agree 😊

Reply:
Ah, I see,- didnt take the context into consideration; thanks for the explanation:)